dir.targets

```diff
@@ -57,6 +57,14 @@
     
     <Error Condition="'$(ErrorIfBuildToolsRestoredFromIndividualProject)'=='true'"
            Text="The build tools package was just restored and so we cannot continue the build of an individual project because targets from the build tools package were not able to be imported. Please retry the build the individual project again." />
+
+    <!--
+      There are cases where the inputs could be newer than the outputs but the
+      download or restore may not need to update. In such cases we need to touch
+      these files otherwise we continually run this target over and over for
+      every project until these files are cleaned (if the are ever cleaned).
+    -->
+    <Touch Files="$(ToolsDir)Microsoft.DotNet.Build.Tasks.dll;$(NugetToolPath)" />
```

Comment:
From my commit message:
The way to enter this state is to keep the packages
directory around for a while and get or make an update to dir.props
or .nuget\packages.config which are the inputs to the target. Then
you will execute the target everytime because nuget.exe nor
the built tools package will actually be updated as part of executing.
To fix this I've added a call to Touch at end of the target to ensure
that the last modified time is always greater on the output files
then the input files.

I'm open to other options here but this seemed to be the most effective way I could figure out how to handle this.